confgenerator/metric_processors.go

```diff
@@ -14,16 +14,182 @@
 
 package confgenerator
 
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
 type MetricsProcessorExcludeMetrics struct {
 	ConfigComponent `yaml:",inline"`
 
 	MetricsPattern []string `yaml:"metrics_pattern,flow" validate:"dive,endswith=/*,startswith=agent.googleapis.com/"`
 }
 
-func (r MetricsProcessorExcludeMetrics) Type() string {
+func (p MetricsProcessorExcludeMetrics) Type() string {
 	return "exclude_metrics"
 }
 
+func (p MetricsProcessorExcludeMetrics) Processors() []otel.Component {
+	var metricNames []string
+	for _, glob := range p.MetricsPattern {
+		// TODO: Remove TrimPrefix when we support metrics with other prefixes.
+		glob = strings.TrimPrefix(glob, "agent.googleapis.com/")
+		var literals []string
+		for _, g := range strings.Split(glob, "*") {
+			literals = append(literals, regexp.QuoteMeta(g))
+		}
+		metricNames = append(metricNames, fmt.Sprintf(`^%s$`, strings.Join(literals, `.*`)))
+	}
+	return []otel.Component{metricsFilter(
+		"exclude",
+		"regexp",
+		metricNames...,
+	)}
+}
+
+// metricsFilter returns an otel.Component that filters metrics.
```

Comment:
Can we split the util stuff below to a separate file? Potentially into the `otel` package?

Reply:
Done, though I did intentionally put it here until the interface was stable and I felt comfortable with exporting it. Oh well, we're still probably going to be the only user of the code.